### PR TITLE
Reland "Dump benchmark flags in CI run" with fix

### PR DESCRIPTION
**Pull request by pzread**
Reland https://github.com/openxla/iree/pull/13649 with fix

Side note: It has been a problem for me to manually test every case of the workflow since we have more and more conditional steps in the workflow... I probably need to read code more carefully, and I'm thinking about automated testing CI workflow.

Prototyped this giant workflow to e2e test 5 kinds of benchmark workflow scenarios: https://github.com/openxla/iree/actions/runs/5017532468 (passed). The code can be found on https://github.com/openxla/iree/pull/13662/commits/6c7fc0df151cd4bff3c739be12b7d05e817d70ef (which turns `ci.yml` into a reusable workflow and calls it with different parameters)

### CODE REVIEW

File: .github/workflows/build_e2e_test_artifacts.yml [hunk 1]
```diff
@@ -122,9 +130,8 @@ jobs:
           # MLIR files and MLIR source models).
           # Not archiving the directory to allow fetching each file as needed
           # separately.
-          find "${E2E_TEST_ARTIFACTS_DIR}" -maxdepth 1 \
-            -name "iree_*" -o -name "model_*.mlir" -o -name "model_*.mlirbc" -o -name "*.json" | \
-            gcloud storage cp --read-paths-from-stdin -r \
+          find "${E2E_TEST_ARTIFACTS_DIR}" -maxdepth 1 ! -name "*.tflite" \
```
Comment (pzread) [inline .github/workflows/build_e2e_test_artifacts.yml]: As we no longer import TF models, `*.tflite` is only the source models we need to exclude from the upload.

**Comment by GMNGeoffrey** [comment]
Can you update the title to say "reland" instead. The double negative is verbose and confusing

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1091,6 +1091,11 @@ jobs:
           <details>
           <summary>Click to show common usages</summary>
 
+          \`\`\`sh
```
Comment (GMNGeoffrey) [inline .github/workflows/ci.yml]: With this amount of text and escaping and such, I think we should consider pulling this out into a template file. It's pretty unclear to me what is getting substituted here. If the substitution can be done only on simple `${VAR}` format, you could use `envsubst` as a lightweight method

**Comment by GMNGeoffrey** [comment]
Regarding testing the CI, this again comes into my "CI optionality" ideas: https://github.com/openxla/iree/issues/10042. I think we want setup to output the list of jobs to run instead of a binary "should-run".

File: .github/workflows/ci.yml (hunk 2 above)
Comment (GMNGeoffrey) [reply .github/workflows/ci.yml]: Can be a follow up

Comment (pzread) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/ARTIFACT_SUMMARY_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+You can list `gs://` directories and download files with [gcloud cli](
+https://cloud.google.com/sdk/gcloud/reference/storage).
+
+<details>
+<summary>Click to show common usages</summary>
+
+```sh
+# Get compile and run flags of benchmarks
+gcloud storage cp -r "${E2E_TEST_ARTIFACTS_GCS_ARTIFACT_DIR}/benchmark-flag-dump.txt" /tmp/
```
Comment (pzread) [inline .github/workflows/ARTIFACT_SUMMARY_TEMPLATE.md:24]: This was `\${E2E_TEST_ARTIFACTS_DIR_URL?}` to demonstrate how to use the variables above. But I found that it is much easier to use by just giving out with the URL.

**Review by GMNGeoffrey — APPROVED** [review]

**Pull request merged**